### PR TITLE
TypeScript Examples

### PR DESCRIPTION
**Pull request by tajo**
All flow (.js) examples now have .tsx counterparts. Visible to users and also integrated with CodeSandbox.

Screen Shot 2019-07-22 at 4 59 18 PM

### CODE REVIEW

File: documentation-site/examples/side-navigation/basic.js [hunk 1]
```diff
@@ -1,3 +1,4 @@
+// @flow
```
Comment (chasestarr) [inline documentation-site/examples/side-navigation/basic.js]:
```suggestion
```

File: scripts/validate-examples.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+*/
+
+/* eslint-env node */
+// @flow
+const fs = require('fs');
+const fsPromises = require('fs').promises;
+const globby = require('globby');
+const path = require('path');
+
+(async () => {
```
Comment (chasestarr) [inline scripts/validate-examples.js]: 🙇

File: documentation-site/examples/side-navigation/basic.js (hunk 1 above)
Comment (chasestarr) [reply documentation-site/examples/side-navigation/basic.js]: saw a couples instances of this

Comment (tajo) [reply documentation-site/examples/side-navigation/basic.js]: yea, it should be in every single `.js` file since we want to enable flow typecheck?

**Comment by tajo** [comment]
> getting `Cannot find module 'styletron-engine-atomic'.ts(2307)` errors when running the code sandbox. is that to be expected?

Not expected. What example is failing?

File: documentation-site/examples/side-navigation/basic.js (hunk 1 above)
Comment (chasestarr) [reply documentation-site/examples/side-navigation/basic.js]: hah oh right

**Comment by chasestarr** [comment]
this link - https://codesandbox.io/s/0swb2?module=/example.tsx (first button example) it took about 15 seconds to display the error

**Comment by tajo** [comment]
> this link - https://codesandbox.io/s/0swb2?module=/example.tsx (first button example) it took about 15 seconds to display the error

Oh right, added missing `@types/styletron-foo` packages. Good catch.

**Pull request merged**